kafka-ui-react-app/src/components/common/Select/Select.tsx

```diff
@@ -1,56 +1,131 @@
-import styled from 'styled-components';
-import React from 'react';
-import { RegisterOptions, useFormContext } from 'react-hook-form';
+import React, { useState, useRef } from 'react';
+import useClickOutside from 'lib/hooks/useClickOutside';
 
-import LiveIcon from './LiveIcon.styled';
 import * as S from './Select.styled';
+import LiveIcon from './LiveIcon.styled';
 
-export interface SelectProps
-  extends React.SelectHTMLAttributes<HTMLSelectElement> {
+export interface SelectProps {
+  options?: Array<SelectOption>;
+  id?: string;
   name?: string;
   selectSize?: 'M' | 'L';
   isLive?: boolean;
-  hookFormOptions?: RegisterOptions;
   minWidth?: string;
+  value?: string | number;
+  defaultValue?: string | number;
+  placeholder?: string;
+  disabled?: boolean;
+  onChange?: (option: string | number) => void;
+}
+
+export interface SelectOption {
+  label: string | number;
+  value: string | number;
+  disabled?: boolean;
 }
 
 const Select: React.FC<SelectProps> = ({
-  className,
-  children,
+  id,
+  options = [],
+  value,
+  defaultValue,
   selectSize = 'L',
+  placeholder = '',
   isLive,
   name,
-  hookFormOptions,
+  disabled = false,
+  onChange,
   ...props
 }) => {
-  const methods = useFormContext();
+  const [selectedOption, setSelectedOption] = useState(value);
+  const [showOptions, setShowOptions] = useState(false);
+
+  const showOptionsHandler = () => {
+    if (!disabled) setShowOptions(!showOptions);
+  };
+
+  const selectContainerRef = useRef(null);
+  const clickOutsideHandler = () => setShowOptions(false);
+  useClickOutside(selectContainerRef, clickOutsideHandler);
+
+  const updateSelectedOption = (option: SelectOption) => {
+    if (disabled) return;
+
+    setSelectedOption(option.value);
+    if (onChange) onChange(option.value);
+    setShowOptions(false);
+  };
+
   return (
-    <div className={`select-wrapper ${className}`}>
+    <div ref={selectContainerRef}>
       {isLive && <LiveIcon />}
       {name ? (
```

Comment:
I have a hard time understanding this condition
Both branches of condition seem identical